### ISSUE TITLE
Log should tell details about the invoice

### ISSUE THREAD

**Issue by douglaz** [fedimint/fedimint#4195]
https://github.com/fedimint/fedimint/blob/58717cb4a08ec89dc4631251a82e96d06a7b9a9d/gateway/ln-gateway/src/lib.rs#L1475

It should tell which invoice it was trying to pay, otherwise it's not very useful.

But we need to consider the comment:
```rust
        // For privacy reasons, we do not return too many details about the failure of
        // the request back to the client to prevent malicious clients from
        // deducing state about the gateway/lightning node.
```
Do you think we could give more details here @m1sterc001guy ?

But anyway this error should log on the gateway side, not only be returned to the client.

Consider also adding the invoice info to the client side of the error:
https://github.com/fedimint/fedimint/blob/58717cb4a08ec89dc4631251a82e96d06a7b9a9d/modules/fedimint-ln-client/src/pay.rs#L253

**Comment by m1sterc001guy** [comment]
> Do you think we could give more details here @m1sterc001guy ?

If we return too many details about the specific error back to the client, the client might be able to deduce certain things about lightning state, like channel balances. 

Definitely agree that we should log more on the gateway side though.

**Comment by elsirion** [comment]
**tl;dr**: yes, let's log the invoice that failed to be paid client-side.

I think the concerns around leaking LN Node state don't apply to logging invoice info at all since that's something the client has access to already.

When logging on the gateway we *might* want to be mindful of user privacy if they are sending whole invoices, but realistically they end up in the LN node DB anyway.
